### DISCUSSION TITLE
error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

### DISCUSSION THREAD

**Discussion by aabdou** [basecamp/kamal#1205] [Q&A]
I set up a user X, added it to the docker group and I can run docker info and docker run successfully when I'm logged in as user X. In my deploy.yml (in my rails app), I have ssh user set to X. When I run kamal setup, I get the error below:
` error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

I'm on Rails 8 beta and my server is running Ubuntu 20.04

**Comment by tuladhar** [comment]
Did you followed these step to manually bootstrap your server?
https://kamal-deploy.org/docs/configuration/ssh/


I'd suggest to try to see if following command works:

```
kamal server exec "whoami" # should show your user X
kamal server exec "docker run hello-world" # should run if docker daemon running and accessible for user X
```

**Comment by aabdou** [comment]
Thank you.

Turned out it was failing locally as it tried to build the container. By
default kamal uses docker-container and on Mac, docker runs in a VM and on
my machine docker-container does not work. I changed Kamal’s config to use
docker instead and it works now :)

**Reply by laptopmutia** [reply]
can u share ur config to fix that?
